### PR TITLE
Rental controller

### PR DESCRIPTION
**Pull request by KyrylH**

### CODE REVIEW

File: src/main/java/com/example/carsharingservice/service/mapper/MapperConfig.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import org.mapstruct.InjectionStrategy;
+import org.mapstruct.NullValueCheckStrategy;
+
+@org.mapstruct.MapperConfig(
+        componentModel = "spring",
+        injectionStrategy = InjectionStrategy.CONSTRUCTOR,
+        nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS
+)
+public interface MapperConfig {
+}
```
Comment (OlehSarapuk) [inline src/main/java/com/example/carsharingservice/service/mapper/MapperConfig.java]: mapper config already exists in main branch

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -1,7 +1,7 @@
-spring.datasource.url=url
+spring.datasource.url=jdbc:mysql://192.168.0.101:32768/carSharingApp?serverTimezone=UTC
 spring.datasource.driverClassName=com.mysql.cj.jdbc.Driver
-spring.datasource.username=name
-spring.datasource.password=password
+spring.datasource.username=root
+spring.datasource.password=732$S8OEp%2o
 spring.jpa.database-platform=org.hibernate.dialect.MySQL8Dialect
```
Comment (OlehSarapuk) [inline src/main/resources/application.properties]: do not push your credentials in github

File: src/main/java/com/example/carsharingservice/controller/RentalController.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import com.example.carsharingservice.service.mapper.RentalMapper;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
```
Comment (OlehSarapuk) [inline src/main/java/com/example/carsharingservice/controller/RentalController.java]: you have not run checkstyle

File: src/main/java/com/example/carsharingservice/controller/RentalController.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    private final CarService carService;
+    private final UserService userService;
+    private final RentalMapper rentalMapper;
+
+    @GetMapping("/{id}")
+    public RentalResponseDto getSpecificRental(@PathVariable Long id) {
+        return rentalMapper.toDto(rentalService.getById(id));
+    }
+
+    @GetMapping("/")
```
Comment (OlehSarapuk) [inline src/main/java/com/example/carsharingservice/controller/RentalController.java]: you can not use ("/") as you have done with @PostMapping

File: src/main/java/com/example/carsharingservice/controller/RentalController.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+    @PostMapping
+    public RentalResponseDto createRental(@RequestBody RentalRequestDto rentalRequestDto) {
+        Car car = carService.getById(rentalRequestDto.getCarId());
+        User user = userService.getById(rentalRequestDto.getUserId());
+        return rentalMapper.toDto(
+                rentalService.createNewRental(car, user, rentalRequestDto.getRentalReturn())
+        );
+    }
+}
```
Comment (OlehSarapuk) [inline src/main/java/com/example/carsharingservice/controller/RentalController.java:63]: add swagger description to all operations and params

**Review by OlehSarapuk — CHANGES_REQUESTED** [review]
Well done!)
Lets improve our project

**Review by OlehSarapuk — APPROVED** [review]

**Review by Nikitos787 — APPROVED** [review]

**Pull request merged**